snap/snapcraft.yaml

```diff
@@ -214,19 +216,35 @@ parts:
       cp -r ./* $SNAPCRAFT_PART_INSTALL
     override-prime: |
       snapcraftctl prime
-      rm -vf usr/lib/jvm/java-17-openjdk-amd64/lib/security/cacerts
       mkdir -p $SNAPCRAFT_PART_INSTALL/opt/kafka
-  prometheus-exporter:
+  extra-maven-libs:
     plugin: nil
     after: [kafka]
     override-build: |
-      curl -o jmx_prometheus_javaagent.jar "https://repo1.maven.org/maven2/io/prometheus/jmx/jmx_prometheus_javaagent/0.17.0/jmx_prometheus_javaagent-0.17.0.jar"
-      EXPECTED_SHA="48ab23a7514f9de5d2ca0acbb8ed1573b3c2ecbef7c5dc4d37c4ba125e538199"
-      JAR_SHA=$(sha256sum jmx_prometheus_javaagent.jar | cut -d " " -f1)
-      if [ $JAR_SHA != $EXPECTED_SHA ]; then exit 1; fi
-      cp jmx_prometheus_javaagent.jar $SNAPCRAFT_PART_INSTALL/
-    organize:
-      jmx_prometheus_javaagent.jar: opt/kafka/jmx_prometheus_javaagent.jar
+      # path jar version sha1 dest
+      jars=(
+        "io/strimzi kafka-oauth-common 0.14.0 90129a72d8c3590e15428f92f2439aebeedf16e4 libs"
+        "io/strimzi kafka-oauth-client 0.14.0 f4ff693ee82f904d51c2d3f7b61b82a5ca87e9da libs"
+        "io/strimzi kafka-oauth-server 0.14.0 163e437aeaa21c85d82598a8bdc417ac9e20be37 libs"
+        "io/strimzi kafka-oauth-server-plain 0.14.0 f57314767738f079b3aabb326a94f63cc4921867 libs"
+        "com/nimbusds nimbus-jose-jwt 9.37.1 940fec997571d75f391afe3988b7fbef7b04b3be libs"
+        "io/prometheus/jmx jmx_prometheus_javaagent 0.17.0 f52f16e90333f2b2e05a47195328a5821b0cd971 opt/kafka"
+      )
```

Comment:
**todo:** Could we move all these jars to `$SNAP/opt/kafka/libs` please?

Reply:
And on that, do we need to update Kafka's `$CLASSPATH` anywhere to use this?
For the Prometheus exporter, we just pass the path in `KAFKA_JMX_OPTS` env-var.

Reply:
> **todo:** Could we move all these jars to `$SNAP/opt/kafka/libs` please?

Is this what the installed snap should look like:
```bash
root@me:/snap/charmed-kafka/current/opt/kafka# ls
bin-wrapper.bash  jmx_prometheus_javaagent-0.17.0.jar  libs  start-wrapper.bash
root@me:/snap/charmed-kafka/current/opt/kafka# ls libs/
kafka-oauth-client-0.14.0.jar  kafka-oauth-common-0.14.0.jar  kafka-oauth-server-0.14.0.jar  kafka-oauth-server-plain-0.14.0.jar  nimbus-jose-jwt-9.37.1.jar
root@me:/snap/charmed-kafka/current/opt/kafka#
```
> And on that, do we need to update Kafka's `$CLASSPATH` anywhere to use this? For the Prometheus exporter, we just pass the path in `KAFKA_JMX_OPTS` env-var.
 
 Yes, we would have to update the $CLASSPATH...

Reply:
or does jmx_prometheus_javaagent-0.17.0.jar also go in opt/kafka/libs?